File: src/main/java/com/vsharstuk/pizzatoppings/util/PageableUtils.java
package com.vsharstuk.pizzatoppings.util;

import lombok.experimental.UtilityClass;
import org.apache.commons.lang3.StringUtils;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;

@UtilityClass
public final class PageableUtils {

    public static Pageable of(Integer pageNumber, Integer pageSize, String sort) {
        if (StringUtils.isNotEmpty(sort)) {
            return of(pageNumber, pageSize, Sort.by(getDirection(sort)));
        }
        return of(pageNumber, pageSize, Sort.by(Sort.Direction.ASC));
    }

    public static Sort.Direction getDirection(String sort) {
        try {
            return Sort.Direction.fromString(sort);
        } catch (Exception e) {
            return Sort.Direction.ASC;
        }
    }

    public static PageRequest of(int page, int size, Sort sort) {
        return size == 0
                ? PageRequest.of(0, Integer.MAX_VALUE, sort)
                : PageRequest.of(page - 1, size, sort);
    }
}

File: src/main/kotlin/com/vsharstuk/pizzatoppings/util/PageableUtils.kt
package com.vsharstuk.pizzatoppings.util

import org.apache.commons.lang3.StringUtils
import org.springframework.data.domain.PageRequest
import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Sort
import org.springframework.data.jpa.domain.JpaSort


object PageableUtils {
    @JvmStatic
    fun of(pageNumber: Int, pageSize: Int, sort: String, propertyName: String?): Pageable {
        return if (StringUtils.isNotEmpty(sort)) {
            buildPageRequest(pageNumber, pageSize, JpaSort.by(getDirection(sort), propertyName))
        } else buildPageRequest(pageNumber, pageSize, JpaSort.by(Sort.Direction.ASC, propertyName))
    }

    @JvmStatic
    fun getDirection(sort: String): Sort.Direction {
        return try {
            Sort.Direction.fromString(sort)
        } catch (e: Exception) {
            Sort.Direction.ASC
        }
    }

    fun buildPageRequest(page: Int, size: Int, sort: Sort): PageRequest {
        return if (size == 0) PageRequest.of(0, Int.MAX_VALUE, sort) else PageRequest.of(page - 1, size, sort)
    }
}
